core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java

```diff
@@ -289,6 +289,10 @@ synchronized void invalidateCache() {
             }
             for (final ReverseBuildTrigger trigger : triggers) {
                 if (trigger.shouldTrigger(r, listener)) {
+                    // Make SpotBugs happy.
```

Comment:
It might be a valid defect, maybe makes sense to investigate or to have a ticket.
Also see https://github.com/jenkinsci/jenkins/blob/master/pom.xml#L114-L117 which defines a VERY permissive strategy for the core

Reply:
> It might be a valid defect, maybe makes sense to investigate or to have a ticket.

No, it is not a valid defect. I explained why in the PR description:

> Looking at the code, this seems like a false positive, because it is wrapped in the conditional `if (trigger.shouldTrigger(r, listener)) {`, and `shouldTrigger` returns `false` if `job` is null. SpotBugs doesn't seem to be able to detect this. To silence the error, I added an explicit null check (which is harmless).

Let me know if there is another resolution you prefer to the above.